webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -573,11 +573,21 @@ var userOverride = {
         var KM_ASK_USER_DETAILS = mckMessageService.checkArray(
             appOptions.askUserDetails
         );
-        var KM_PRELEAD_COLLECTION = mckMessageService.checkArray(
-            appOptions.preLeadCollection
-        )
-            ? appOptions.preLeadCollection
-            : [];
+        var KM_PRELEAD_COLLECTION;
+        if (appOptions.preLeadCollection) {
+            KM_PRELEAD_COLLECTION = mckMessageService.checkArray(
+                appOptions.preLeadCollection
+            );
+        } else if (
+            appOptions.appSettings.leadCollection &&
```

Comment:
move appOptions.appSettings.collectLead first

Reply:
appOptions.preLeadCollection contain array which is given in installation script.

Reply:
change conditions order, as collectLead is a boolean so better to have it as the first condition in the logical AND
appOptions.appSettings.collectLead and appOptions.appSettings.leadCollection